### PR TITLE
UserProfile Page: Render Description

### PR DESCRIPTION
**Pull request by 6543**
now you can use Markdown to get something like this:
Bildschirmfoto zu 2020-08-03 13-24-37

### CODE REVIEW

**Comment by 6543** [comment]
It's called Biography in Settings but internaly named Description everywhere

**Review by a1012112796 — APPROVED** [review]
Good feature, I like it. Thanks. 🎉
After this change suggest adding simpleMDE support for description edit box.

**Comment by silverwind** [comment]
If this is supposed to mimic GitHub's recent profile README's, I suggest putting it into a box on the right instead of cramming it into that small box.

> simpleMDE

Please don't for now. We want to get rid of SimpleMDE, not make it harder to migrate off it.

**Review by lunny — APPROVED** [review]

**Comment by 6543** [comment]
@silverwind now it only render if Description has text in it

**Comment by 6543** [comment]
@lunny @a1012112796 hope you dont mind the change after your LGTM?

**Comment by a1012112796** [comment]
> Please don't for now. We want to get rid of SimpleMDE, not make it harder to migrate off it.

I see, it's reasonable. I think maybe we can add some note messages on the edit box to let users know the user's description support markdown rules.

**Comment by silverwind** [comment]
I guess this is fine for now, our user profile page will need a rework anyways to allow adding one more tab to it (e.g. create a "Overview" tab and move both profile readme and activity to it).

**Review by silverwind — APPROVED** [review]

**Comment by zeripath** [comment]
@6543 I think the tests are genuinely failing here and don't just represent intermittent failures.

**Comment by 6543** [comment]
look's like I'll have to dick into it

**Comment by 6543** [comment]
Interestint throu that the test fail on points where this PR should **NOT** produce **ANY** change ...

File: routers/user/profile.go [hunk 1]
```diff
@@ -97,6 +98,10 @@ func Profile(ctx *context.Context) {
 	// so everyone would get the same empty heatmap
 	ctx.Data["EnableHeatmap"] = setting.Service.EnableUserHeatmap && !ctxUser.KeepActivityPrivate
 	ctx.Data["HeatmapUser"] = ctxUser.Name
+	if len(ctx.User.Description) != 0 {
```
Comment (zeripath) [inline routers/user/profile.go]:
```suggestion
	if ctx.User != nil && len(ctx.User.Description) != 0 {
```

File: routers/user/profile.go [hunk 2]
```diff
@@ -97,6 +98,10 @@ func Profile(ctx *context.Context) {
 	// so everyone would get the same empty heatmap
 	ctx.Data["EnableHeatmap"] = setting.Service.EnableUserHeatmap && !ctxUser.KeepActivityPrivate
 	ctx.Data["HeatmapUser"] = ctxUser.Name
+	if ctx.User != nil && len(ctx.User.Description) != 0 {
```
Comment (6543) [inline routers/user/profile.go]:
```suggestion
	if len(ctxUser.Description) != 0 {
```

Comment (lunny) [inline routers/user/profile.go]: So user must login to see the profile?
`ctx.User` is the login user, `cxtuser` is the user who has the profile?

**Review by lunny — CHANGES_REQUESTED** [review]

**Comment by 6543** [comment]
☝️ thats how bug's are produced - in this case the nil exeption didnt let me sleep - catched it

**Comment by 6543** [comment]
@lunny you noticed it at the same time :D

**Review by lunny — APPROVED** [review]

**Review by zeripath — APPROVED** [review]

File: routers/user/profile.go (hunk 2 above)
Comment (zeripath) [reply routers/user/profile.go]: Good catch @lunny sorry I missed that

**Pull request merged**